src/components/tooltip/TrackPower.js

```diff
@@ -63,11 +65,40 @@ class TooltipTrackPowerImpl extends React.PureComponent<Props> {
     return sum * 1e-12;
   }
 
+  _computeCO2eSumForRange(start: Milliseconds, end: Milliseconds): number {
```

Comment:
You can probably take the power information as a parameter instead of start/end. You can call it `computeCO2eFromPower` for example.

Then you only need to do the simple multiplication. And then you can use it directly without all the memoization stuff.